### PR TITLE
Add `getparams` and `setparams!!` following AbstractMCMC v5.5 and v5.6

### PR DESCRIPTION
**Pull request by sunxd3**
ref: https://github.com/TuringLang/Turing.jl/issues/2367

### CODE REVIEW

**Comment by sunxd3** [comment]
the failed test is not related to this PR, @yebai maybe you already have an answer to why it fails?

File: src/abstractmcmc.jl [hunk 1]
```diff
@@ -30,6 +30,15 @@ getadaptor(state::HMCState) = state.adaptor
 getmetric(state::HMCState) = state.metric
 getintegrator(state::HMCState) = state.κ.τ.integrator
 
+function AbstractMCMC.getparams(state::HMCState)
+    # TODO(sunxd): should we return a copy?
+    return state.transition.z.θ
+end
+
+function AbstractMCMC.setparams!!(state::HMCState, θ)
```
Comment (torfjelde) [inline src/abstractmcmc.jl:38]: One dangerous aspect there is that `state.transition` _caches_ the logjoint and gradient computations. So when you use the `@set` macro here, you're going to also keep the cached log-joint and gradient computation, which will then be out of sync with the parameters.

If you then naively pass this transition somewhere, say, into the next `step` call, AHMC.jl will use the incorrect logjoint eval in the MH step.

IMO the safe way is to use the explicit constructor of `PhasePoint` I believe without passing in the cached values. This should result in receomputation of this.

**Review by torfjelde — CHANGES_REQUESTED** [review]
The current impl is going to lead to incorrect log-density values being used in the transition. IMO we shouuld do the safe thing by default here.

File: src/abstractmcmc.jl (hunk 1 above)
Comment (sunxd3) [reply src/abstractmcmc.jl]: just realized that `model` is not passed into `setparams!!`, I think for now, we only set the parameters, then when use the transition, logp should be recomputed. (we can also later introduce some like `setlogp` or `compute_logp` etc.) I'll add a comment in the code

**Comment by sunxd3** [comment]
let's delay a bit to see through https://github.com/TuringLang/AbstractMCMC.jl/pull/150

**Review by torfjelde — CHANGES_REQUESTED** [review]
The fix to `setparams!!` doesn't work unfortauntely. We need access to the `model` here, which we don't have.

IMO we should add `model` as an argument to the AbstractMCMC.jl interface. Otherwise there's no way we can do stuff like update the gradient information, which is an issue we're facing both here and in AdvancedMH.jl.

**Review by torfjelde — APPROVED** [review]
Good stuff @sunxd3 :)

**Pull request merged**